veps/sig-compute/decentralized-migration/decentralized-migration-vep.md

```diff
@@ -0,0 +1,279 @@
+
+The target cluster will have a synchronization controller running in the kubevirt namespace, that has access to the migration channel (either pod network, or migration network). This controller waits for incoming synchronization connections from a source. When a `source` migration resource is created in the cluster, this synchronization controller will connect to the target synchronization and start synchronizing the `VirtualMachineInstance` status between them.
+
+If the source and the target are in the same cluster, the connection will be inside of the synchronization controller.
+
+Once synchronization has been established, each `virt-handler` pod will interact with their respective `VirtualMachineInstance` the same way as before. The major difference being that it is possible there are two separate `VirtualMachineInstance`s.
+
+TBD: How to handle the source `VirtualMachine` once the migration has completed. We can shut down the source `VirtualMachine`, delete it, or keep it running. For now a good starting point is to simply shut down the `VirtualMachine` and someone else decide what to do with the `VirtualMachine`
+
+Figure 2 describes the interaction with a source and target `VirtualMachineInstanceMigration` in the same namespace
+
+![figure 2](decentralized_in_namespace.png)
+
+Figure 3 describes the interaction with a source and target `VirtualMachineInstanceMigration` in different namespaces.
+
+![figure 3](decentralized_cross_namespace.png)
+
+Figure 4 describes the interaction with a source and target `VirtualMachineInstanceMigration` in different clusters
+
+![figure 4](decentralized_cross_cluster.png)
+
+#### Synchronization controller
+The synchronization controller allows one to synchronize two distinct and separate `VirtualMachineInstance`s. The synchronization controller is a separate controller that has access to modify `VirtualMachineInstance`s and can connect to the migration network.
+
+There is a single active synchronization controller per cluster that listens on an established address port combination. TBD: how to advertise this address (maybe kubevirt CR?).
+
+The `VirtualMachineInstanceMigration` resource will contain the name of the `VirtualMachineInstance` and a mapping key. Both the `source` and `target` will use the same mapping key, so the synchronization controller can map the received `VirtualMachineInstance` to its respective local instance.
+
+Synchronization is achieved by following the flows in figure 5 and 6
+##### Incoming VMI
+
+![figure 5](sync_controller_incoming_vmi.png)
+##### Reconcile loop
+![figure 6](sync_controller_reconcile.png)
+
+## API Examples
+
+Two major changes to the API.
+
+#### VirtualMachineInstanceMigration
+
+A new `key` field that can be any unique string. This is an optional field. If it is omited this means it standard migration just like it exists now. The suggested unique id, is either the UID of the source VM or the VM of the target VM.
```

Comment:
What is the purpos eof the key?

Reply:
The key is so we can unique identify the migration. Assuming the synchronzition controller handles multiple migrations at once.